src/lsm/schema.zig

```diff
@@ -509,3 +520,122 @@ pub const TableData = struct {
         return schema.block_values_bytes_const(data_block)[0..used_bytes];
     }
 };
+
+/// A ManifestLog block's body is a SoA of Labels and TableInfos.
+/// Each Label/TableInfo pair is an "entry".
+// TODO Store timestamp (snapshot) in header.
+pub const ManifestLog = struct {
+    const entry_size = @sizeOf(Label) + @sizeOf(TableInfo);
+    const entry_count_max_unaligned = @divFloor(block_body_size, entry_size);
+
+    pub const entry_count_max =
+        @divFloor(entry_count_max_unaligned, @alignOf(TableInfo)) * @alignOf(TableInfo);
```

Comment:
Alternatively, maybe layout stuff in the order of decreasing alignment? There won't be any gaps that way, so no adjustments are necessary. 

I think this migh be a good rule-of-thumb for all on-disk SoA layouts?

Reply:
> Alternatively, maybe layout stuff in the order of decreasing alignment? There won't be any gaps that way, so no adjustments are necessary.

That's generally a good rule. However, sometimes we trump that intentionally. For example, packing fields of a `user_data/reserved` nature together contiguously so that they can be modified by flags in future and regrouped, or to put "well known" fields first.

Reply:
> I think I am slightly in favor of maybe changeing the layout to make all arrays varlength, so it's more unfiform.

I think the gotcha here is that we construct manifest log blocks incrementally. Right now we can do that in-place with no extra copies (i.e. writing the label and entry directly to the memory of the block we are building), since the two arrays (labels and tableinfos) are in fixed positions/offsets within the block body. But if the labels array (or whichever array comes first) is variable-length, then that breaks down.

Reply:
(Not directly related to this thread, but I think I will directly store the `entry_count: u32` in the manifest log block header's `context` field, rather than deriving it from `size`.)

Reply:
Ahh, yeah. I guess this could be added as a comment to motivate the current layout? 

But maybe we could do this:

* use vararray soa layout in the order of decrease alignment. 
* construct the block incrementally, _assuming that its going to be full_. 
* when the block is finished and it is not full, `copy_left` to compact it before writing